operator/internal/manifests/storage/schema_test.go

```diff
@@ -0,0 +1,613 @@
+package storage
+
+import (
+	"testing"
+	"time"
+
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BuildSchemaConfigList_NoSchemas(t *testing.T) {
+	specs := []lokiv1beta1.ObjectStorageSchemaSpec{}
+	statuses := []lokiv1beta1.StorageSchemaStatus{}
+
+	_, err := BuildSchemaConfigList(time.Now().UTC(), specs, statuses)
+
+	require.Error(t, err)
+}
+
+func BuildSchemaConfigList_AddSchema_NoStatuses(t *testing.T) {
+	specs := []lokiv1beta1.ObjectStorageSchemaSpec{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+	}
+	statuses := []lokiv1beta1.StorageSchemaStatus{}
+
+	actual, err := BuildSchemaConfigList(time.Now().UTC(), specs, statuses)
+	expected := []lokiv1beta1.ObjectStorageSchemaSpec{
+		{
+			Version:       lokiv1beta1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+	}
+
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+func BuildSchemaConfigList_AddSchema_WithPriorSchemas_OutsideUpdateWindow(t *testing.T) {
```

Comment:
nit. `Outside` can have ambiguous meaning, i.e.:
- It can mean outside before time window
- It can mean outside after time window

I suggest to rename this to switch to something `Valid/Invalid` or `Matching/Conflicting`.